### PR TITLE
Update Glossary with new roles and policy

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
### Describe your changes :

I worked on the supporting new permission for the glossary page.

#
### Type of change :

- [x] New feature

#
### Frontend Preview (Screenshots) :

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@open-metadata/ui

### CODE REVIEW

**Review by aniketkatkar97 — APPROVED** [review]

**Review by ShaileshParmar11 — APPROVED** [review]

**Comment by ShaileshParmar11** [comment]
Add glossary button is disabled that's why the cypress is failing can you check?

File: openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryV1.component.tsx [hunk 1]
```diff
@@ -327,32 +401,41 @@ const GlossaryV1 = ({
...
+            overlay={manageButtonContent()}
+            overlayStyle={{ width: '350px' }}
+            placement="bottomRight"
+            trigger={['click']}
+            visible={showActions}
+            onVisibleChange={setShowActions}>
+            <Tooltip
+              title={
+                glossaryPermission.Delete
+                  ? 'Manage Glossary'
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryV1.component.tsx:432]: Is this text correct?

Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryV1.component.tsx:432]: Yes, There we have a button with three vertical dots.
image

Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryV1.component.tsx:433]: So we should't disable it based on Delete permission..
We can disable delete option in it but not the more option based on delete permission

Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryV1.component.tsx:433]: Okay, got it!.

**Pull request merged**